Sources/Host/POSIX/File.cpp

```diff
@@ -18,8 +18,36 @@
 namespace ds2 {
 namespace Host {
 
+static uint32_t convertFlags(uint32_t ds2Flags) {
+  uint32_t flags = 0;
+
+  if (ds2Flags & kModeRead)
+    flags |= (ds2Flags & kModeWrite) ? O_RDWR : O_RDONLY;
+  else if (ds2Flags & kModeWrite)
+    flags |= O_WRONLY;
+
+  if (ds2Flags & kModeAppend)
+    flags |= O_APPEND;
+  if (ds2Flags & kModeTruncate)
+    flags |= O_TRUNC;
+  if (ds2Flags & kModeNonBlocking)
+    flags |= O_NONBLOCK;
+  if (ds2Flags & kModeCreate)
+    flags |= O_CREAT;
+  if (ds2Flags & kModeNewOnly)
+    flags |= O_EXCL;
+#if defined(OS_LINUX)
```

Comment:
Here you should probably do `#if defined(O_NOFOLLOW)` instead, because I suppose systems other than Linux support that flag.

Also, adding an `#else` block and returning an error like unsupported or invalidargument *might* make sense?